### PR TITLE
snow-85722 Perform check for thread index allocation and free index

### PR DESCRIPTION
**Pull request by ChTimTsubasa**
https://snowflakecomputing.atlassian.net/browse/SNOW-85722

### CODE REVIEW

File: cpp/util/ThreadPool.hpp [hunk 1]
```diff
@@ -143,6 +145,12 @@ class ThreadPool {
...
 #ifdef _WIN32
     key = TlsAlloc();
+    if (key == TLS_OUT_OF_INDEXES)
+    {
+      CXX_LOG_ERROR("Thread pool out of TLS index");
+      throw SnowflakeTransferException(TransferError::INTERNAL_ERROR,
+                                               "Out of TLS index in the thread pool");
+    }
 #else
     pthread_key_create(&key, NULL);
```
Comment (rvishureddy) [inline cpp/util/ThreadPool.hpp]: Can you also do the check for Linux platform. pthread_key_create fails when the system has insufficient memory or if max keys is reached.

File: tests/test_connect_negative.c [hunk 2]
```diff
@@ -121,9 +121,9 @@ int main(void) {
       cmocka_unit_test(test_connect_account_missing),
       cmocka_unit_test(test_connect_user_missing),
       cmocka_unit_test(test_connect_password_missing),
-      cmocka_unit_test(test_connect_invalid_database),
-      cmocka_unit_test(test_connect_invalid_schema),
-      cmocka_unit_test(test_connect_invalid_warehouse),
+      //cmocka_unit_test(test_connect_invalid_database),
```
Comment (notkriswagner) [inline tests/test_connect_negative.c]: Can you create a JIRA to re-enable these tests and assign it to me for the next sprint?

File: tests/test_connect_negative.c [hunk 3]
```diff
@@ -121,9 +121,9 @@ int main(void) {
...
+      //cmocka_unit_test(test_connect_invalid_schema),
+      //cmocka_unit_test(test_connect_invalid_warehouse),
       cmocka_unit_test(test_connect_invalid_role),
```
Comment (rvishureddy) [inline tests/test_connect_negative.c]: Ok to comment out these tests created SNOW-85815 to track.

File: cpp/util/ThreadPool.hpp (hunk 1 above)
Comment (ChTimTsubasa) [reply cpp/util/ThreadPool.hpp]: Good point. I will add it.

File: tests/test_connect_negative.c (hunk 2 above)
Comment (ChTimTsubasa) [reply tests/test_connect_negative.c]: @rvishureddy created it and I CCed you on SNOW-85815

**Comment by howryu** [comment]
So what do we do if tls alloc failed? We just throw exception to client directly?

**Pull request merged**